### ISSUE TITLE
a11y standards for Labs projects

### ISSUE THREAD

**Issue by rholdy** [Lambda-School-Labs/labs-engineering-standards#22]

**Comment by berniedurfee** [comment]
We could require running something like https://github.com/dequelabs/react-axe during the CI process as a check before merge to master.

https://github.com/dequelabs/axe-cli

https://codeburst.io/automated-accessibility-testing-tool-a11y-pa11y-jest-storybook-2ad294bfe71a

**Comment by berniedurfee** [comment]
I think we need a 'Linting' topic that covers what linting (static code analysis in general) is required in different scenarios.

First pass would be to require passing runs of:
- Code style
- Accessibility

We should explicitly declare the tools and the rules that need to be enabled within those tools.

**Comment by jgmuchiri** [comment]
@berniedurfee is [this the initial](https://lambda-school-labs.github.io/labs-engineering-standards/topics/writing-standards/#linting) pass for this topic?

**Comment by JimmyMcBride** [comment]
I think linting is a great idea. ESLint is pretty popular and a lot of people. It works with JS and TS projects alike. Airbnb rules are pretty popular as well. What kind of linting rules and linter are you thinking about, Bernie?

**Comment by berniedurfee** [comment]
> @berniedurfee is [this the initial](https://lambda-school-labs.github.io/labs-engineering-standards/topics/writing-standards/#linting) pass for this topic?

No, that's just linting for the Markdown using for this standards document. This would be requiring linting on application code. It would need to be enforced via a CI script or by Code Climate.

**Comment by berniedurfee** [comment]
> I think linting is a great idea. ESLint is pretty popular and a lot of people. It works with JS and TS projects alike. Airbnb rules are pretty popular as well. What kind of linting rules and linter are you thinking about, Bernie?

I always favor starting small and adding as we go. So _very_ basic linting using the core rules in ESLint would be a good start.

The big issue is that I hate to ratify a standard that we can't monitor compliance for at scale. We _could_ do this wil Code Climate, but their API doesn't allow for programmatic configuration of linting... so I (being the only admin, because there can only be one) would need to manually set this up for every repo... that doesn't scale.

Let's hold off on this. I'm going to open a ticket with Code Climate to see if there's a way to enable ESLint checking via automation. If not, maybe we can come up with an alternative solution using GitHub actions.
